config/vite.php

```diff
@@ -2,21 +2,10 @@
 
 use craft\helpers\App;
 
-
-// Decides which port to use for devServerPublic. This allows
-// you to visit the http or https port. If you plant to only
-// use http, make sure that PRIMARY_SITE_URL is set to http
-//
-// Match ports to .ddev/config.yaml -> web_extra_exposed_ports
-$host = Craft::$app->getRequest()->getIsConsoleRequest()
-    ? App::env('PRIMARY_SITE_URL')
-    : Craft::$app->getRequest()->getHostInfo();
-$httpPort = 3000;
-$httpsPort = 3001;
-$devServerPort = str_starts_with($host, 'https') ? $httpsPort : $httpPort;
-
 return [
-    'devServerPublic' => "$host:$devServerPort",
+    // Match to https port in .ddev/config.yaml -> web_extra_exposed_ports
+    // Only works when using HTTPS on DDEV.
+    'devServerPublic' => App::env('PRIMARY_SITE_URL') . ':3001',
```

Comment:
~Does it make sense to use the variable above to both be more explicit and DRY?~

```suggestion
    'devServerPublic' => App::env('PRIMARY_SITE_URL') . ':3000', // matches ddev web_extra_exposed_ports.https_port
```

Reply:
Oh wait, now I see you simplified further than I realized. I don't think we should take out the note about the ports needing to match web_extra_exposed_ports. Let me re-read this PR

Reply:
Here's what I have working on another site:
`.ddev/config.yaml`
```
web_extra_exposed_ports:
  # Expose ports for vite
  - name: vite
    container_port: 3000
    http_port: 3001
    https_port: 3000
```

`config/vite.php`
```
<?php

use craft\helpers\App;

/*
 $host = Craft::$app->getRequest()->getIsConsoleRequest()
    ? App::env('PRIMARY_SITE_URL')
    : Craft::$app->getRequest()->getHostInfo();
*/
$host = App::env('PRIMARY_SITE_URL');

/*
.ddev/config.yaml
    container_port: 3000
    http_port: 3001
    https_port: 3000
 */

$httpPort = 3001;
$httpsPort = 3000;
$devServerPort = str_starts_with($host, 'https') ? $httpsPort : $httpPort;

return [
    'checkDevServer' => true,
    'devServerInternal' => 'http://localhost:' . $httpsPort,
    'devServerPublic' => preg_replace('/:\d+$/', '', App::env('PRIMARY_SITE_URL')) . ':' . $httpsPort,
    'serverPublic' => App::env('PRIMARY_SITE_URL') . '/dist/',
    'useDevServer' => App::env('ENVIRONMENT') === 'dev' || App::env('CRAFT_ENVIRONMENT') === 'dev',
    'manifestPath' => '@webroot/dist/manifest.json', // Vite 4
    'includeReactRefreshShim' => false,
    'includeModulePreloadShim' => false,
    'errorEntry' => '',
    'cacheKeySuffix' => '',
];
```

`vite.config.js`
```
...
server: {
      allowedHosts: ['.ddev.site'],
      host: '0.0.0.0',
      port: HTTPS_PORT,
      strictPort: true,
      origin: `${process.env.DDEV_PRIMARY_URL?.replace(/:\d+$/, '')}:${HTTPS_PORT}`,
      cors: {
        origin: /https?:\/\/([A-Za-z0-9\-\.]+)?(\.ddev\.site)(?::\d+)?$/,
      },
    },
...
```